test/code/plugins/squid_log_parser_plugintest.rb

```diff
@@ -0,0 +1,150 @@
+## squid_log_parser_plugintest
+## Created by Alessandro Cardoso
+## 
+
+
+#require 'fluent'
+#require 'fluent/parser'
+require 'test/unit'
+require_relative '../../../source/code/plugins/squidlogparser.rb'
+
+class SquidLogParserTestRuntimeError < SquidLogParser      #Fluent::SquidLogParser
+  def log_error(text)
```

Comment:
Yes, but the [auditlog_lib.rb](https://github.com/Microsoft/OMS-Agent-for-Linux/blob/master/source/code/plugins/auditlog_lib.rb#L7) overrides the LoggingBase type. You don't do this, so you don't need to override it in your test. Eliminating this may fix the errors.